src/psbt/mod.rs

```diff
@@ -37,15 +46,35 @@ impl PsbtUtils for Psbt {
             None
         }
     }
+
+    fn fee_amount(&self) -> u64 {
```

Comment:
I'm wondering why the `output_amount` is calculated using the `unsigned_tx` outputs and not the PSBT `outputs`.

Reply:
The PSBT and transactions inputs and outputs should all be the same. But your questions made me think about moving these functions to the TransactionDetails struct, or maybe as new util trait on Transaction. Will noodle around with that and see how it works out.

Reply:
Also I just remembered the reason I needed to use PSBT to calculate the `fee_amount` is because it contains the full details for the transaction inputs, which includes the input amounts, if you only look at the Transaction structure it contains essentially "pointers" to the unspent outputs that become the inputs (the `OutPoint`) and this structure does not contain the input amounts.  So I'm leaving these helper functions on the `PsbtUtils` trait.